sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -2378,6 +2378,8 @@ class DataSourceV2SQLSuiteV1Filter
     val t = "testcat.ns1.ns2.tbl"
     withTable(t) {
       spark.sql(s"CREATE TABLE $t (id bigint, data string) USING foo")
+      checkAnswer(sql(s"SHOW COLUMNS FROM $t IN testcat.ns1.ns2"), Seq(Row("id"), Row("data")))
```

Comment:
Are the changes in these two lines related to the current PR?

Reply:
Yes，from the last pr https://github.com/apache/spark/pull/47568， we lost the two line check，so this pr add these.

Reply:
I apologize for not having a clear understanding of the background before and providing misleading guidance. 

If this is a followup to SPARK-49078, then indeed we should only clean up `showColumnsNotSupportedForV2TablesError` in this PR and add these two new test checks. 

However, for these two new test checks, we should modify the PR title and description to make them more relevant. 

For other unused methods in `QueryCompilationErrors`, they should be addressed in a separate PR.

Reply:
OK，no problem，would make a separate pr to do clean other case，then this pr the same with last？ WDYT @LuciferYang

Reply:
Had recover as before~ @LuciferYang